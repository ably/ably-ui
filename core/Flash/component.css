@layer components {
  .ui-flash {
    @apply w-full fixed;
    top: 5.5rem;
    z-index: calc(var(--stacking-context-page-meganav) - 10);
    transition: margin-top 200ms;
  }

  .ui-flash-message {
    @apply font-sans font-light antialiased max-w-screen-xl mx-auto mt-8 opacity-0 relative;
    transition: opacity 200ms, transform 200ms, height 200ms 200ms,
      margin-top 200ms 200ms;
    transform: translateY(-200%) rotateX(-90deg);
  }

  .ui-flash-text {
    @apply mr-16 text-p1;
  }

  /* dynamic content inside flash, can't add classes */
  .ui-flash-text a {
    @apply underline;
  }

  .ui-flash-message-enter {
    @apply opacity-100;
    transform: translateY(0) rotateX(0);
  }
}


/*# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIndlYnBhY2s6Ly9BYmx5VWkuQ29yZS5bbmFtZV0vLi9zcmMvY29yZS9GbGFzaC9jb21wb25lbnQuY3NzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiJBQUFBO0VBQ0U7SUFDRSxtQkFBbUI7SUFDbkIsV0FBVztJQUNYLHdEQUF3RDtJQUN4RCw0QkFBNEI7RUFDOUI7O0VBRUE7SUFDRSx1RkFBdUY7SUFDdkY7NEJBQ3dCO0lBQ3hCLDRDQUE0QztFQUM5Qzs7RUFFQTtJQUNFLG9CQUFvQjtFQUN0Qjs7RUFFQSxvREFBb0Q7RUFDcEQ7SUFDRSxnQkFBZ0I7RUFDbEI7O0VBRUE7SUFDRSxrQkFBa0I7SUFDbEIsbUNBQW1DO0VBQ3JDO0FBQ0YiLCJmaWxlIjoiY29yZS9GbGFzaC9jb21wb25lbnQuY3NzIiwic291cmNlc0NvbnRlbnQiOlsiQGxheWVyIGNvbXBvbmVudHMge1xuICAudWktZmxhc2gge1xuICAgIEBhcHBseSB3LWZ1bGwgZml4ZWQ7XG4gICAgdG9wOiA1LjVyZW07XG4gICAgei1pbmRleDogY2FsYyh2YXIoLS1zdGFja2luZy1jb250ZXh0LXBhZ2UtbWVnYW5hdikgLSAxMCk7XG4gICAgdHJhbnNpdGlvbjogbWFyZ2luLXRvcCAyMDBtcztcbiAgfVxuXG4gIC51aS1mbGFzaC1tZXNzYWdlIHtcbiAgICBAYXBwbHkgZm9udC1zYW5zIGZvbnQtbGlnaHQgYW50aWFsaWFzZWQgbWF4LXctc2NyZWVuLXhsIG14LWF1dG8gbXQtOCBvcGFjaXR5LTAgcmVsYXRpdmU7XG4gICAgdHJhbnNpdGlvbjogb3BhY2l0eSAyMDBtcywgdHJhbnNmb3JtIDIwMG1zLCBoZWlnaHQgMjAwbXMgMjAwbXMsXG4gICAgICBtYXJnaW4tdG9wIDIwMG1zIDIwMG1zO1xuICAgIHRyYW5zZm9ybTogdHJhbnNsYXRlWSgtMjAwJSkgcm90YXRlWCgtOTBkZWcpO1xuICB9XG5cbiAgLnVpLWZsYXNoLXRleHQge1xuICAgIEBhcHBseSBtci0xNiB0ZXh0LXAxO1xuICB9XG5cbiAgLyogZHluYW1pYyBjb250ZW50IGluc2lkZSBmbGFzaCwgY2FuJ3QgYWRkIGNsYXNzZXMgKi9cbiAgLnVpLWZsYXNoLXRleHQgYSB7XG4gICAgQGFwcGx5IHVuZGVybGluZTtcbiAgfVxuXG4gIC51aS1mbGFzaC1tZXNzYWdlLWVudGVyIHtcbiAgICBAYXBwbHkgb3BhY2l0eS0xMDA7XG4gICAgdHJhbnNmb3JtOiB0cmFuc2xhdGVZKDApIHJvdGF0ZVgoMCk7XG4gIH1cbn1cbiJdLCJzb3VyY2VSb290IjoiIn0=*/