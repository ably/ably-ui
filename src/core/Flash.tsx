import React, { useEffect, useState, useRef } from "react";
import DOMPurify from "dompurify";
import { getRemoteDataStore } from "./remote-data-store.js";
import ConnectStateWrapper from "./ConnectStateWrapper";
import Icon from "./Icon";
import { ColorClass } from "./styles/colors/types";
import { IconName } from "./Icon/types";

type FlashPropsType = "error" | "success" | "notice" | "info" | "alert";

type FlashProps = {
  id: string;
  removed: boolean;
  type: FlashPropsType;
  content: string;
  removeFlash: (id: string) => void;
};

type FlashesProps = {
  flashes: { items: FlashProps[] };
};

type BackendFlashesProps = {
  flashes: string[][];
};

const REDUCER_KEY = "flashes";
const FLASH_DATA_ID = "ui-flashes";

const initialState = { items: [] };

const reducerFlashes = {
  [REDUCER_KEY]: (
    state: {
      items: FlashProps[];
    } = initialState,
    action: { type: string; payload: FlashProps | FlashProps[] },
  ) => {
    switch (action.type) {
      case "flash/push": {
        const flashes = Array.isArray(action.payload)
          ? action.payload
          : [action.payload];
        return { items: [...state.items, ...flashes] };
      }
      default:
        return state;
    }
  },
};

// Not cool but redux isn't a long term plan here
// eslint-disable-next-line @typescript-eslint/no-explicit-any
const selectFlashes = (store: any): { items: FlashProps[] } =>
  store.getState()[REDUCER_KEY];

const FLASH_BG_COLOR = {
  error: "bg-gui-error",
  success: "bg-zingy-green",
  notice: "bg-electric-cyan",
  info: "bg-electric-cyan",
  alert: "bg-active-orange",
};

const FLASH_TEXT_COLOR = {
  error: "text-white",
  success: "text-cool-black",
  notice: "text-cool-black",
  info: "text-cool-black",
  alert: "text-white",
};

const AUTO_HIDE = ["success", "info", "notice"];
const AUTO_HIDE_TIME = 8000;

const useAutoHide = (type: string, closeFlash: () => void) => {
  const timeoutId = useRef<ReturnType<typeof setTimeout> | null>(null);

  useEffect(() => {
    if (AUTO_HIDE.includes(type)) {
      timeoutId.current = setTimeout(() => {
        closeFlash();
      }, AUTO_HIDE_TIME);
    }

    return () => {
      if (timeoutId.current) {
        clearTimeout(timeoutId.current);
      }
    };
  }, []);
};

const Flash = ({ id, type, content, removeFlash }: FlashProps) => {
  const ref = useRef<HTMLDivElement>(null);
  const [closed, setClosed] = useState(false);
  const [flashHeight, setFlashHeight] = useState(0);
  const [triggerEntryAnimation, setTriggerEntryAnimation] = useState(false);

  const closeFlash = () => {
    if (ref.current) {
      setFlashHeight(ref.current.getBoundingClientRect().height);
    }

    setClosed(true);

    setTimeout(() => {
      id && removeFlash(id);
    }, 100);
  };

  useEffect(() => setTriggerEntryAnimation(true), []);
  useAutoHide(type, closeFlash);

  const animateEntry = triggerEntryAnimation && !closed;

  let style;

  if (flashHeight && !closed) {
    style = { height: `${flashHeight}px` };
  } else if (closed) {
    style = { height: 0, marginTop: 0, zIndex: -1 };
  } else {
    style = {};
  }

  const safeContent = DOMPurify.sanitize(content, {
    ALLOWED_TAGS: ["a"],
    ALLOWED_ATTR: ["href", "data-method", "rel"],
  });

  const withIcons: Record<FlashPropsType, IconName | ""> = {
    notice: "icon-gui-ably-badge",
    success: "icon-gui-tick",
    error: "icon-gui-warning",
    alert: "icon-gui-warning",
    info: "",
  };

  const iconColor: Record<FlashPropsType, ColorClass | ""> = {
    notice: "text-cool-black",
    success: "text-cool-black",
    error: "text-white",
    alert: "text-white",
    info: "",
  };

  return (
    <div
      className={`ui-flash-message ui-grid-px ${
        animateEntry ? "ui-flash-message-enter" : ""
      }`}
      style={style}
      ref={ref}
      data-id="ui-flash"
    >
      <div
        className={`${FLASH_BG_COLOR[type]} p-32 flex align-center rounded shadow-container-subtle`}
      >
        {withIcons[type] && iconColor[type] && (
          <Icon
            name={withIcons[type]}
            color={iconColor[type]}
            size="1.5rem"
            additionalCSS="mr-16 self-baseline"
          />
        )}
        <p
          className={`ui-text-p1 mr-16 ${FLASH_TEXT_COLOR[type]}`}
          dangerouslySetInnerHTML={{ __html: safeContent }}
        />
        <button
          type="button"
          className="p-0 ml-auto self-start focus:outline-none"
          onClick={closeFlash}
        >
          {iconColor[type] && (
            <Icon
              name="icon-gui-close"
              color={iconColor[type]}
              size="1.5rem"
              additionalCSS="transition-colors"
            />
          )}
        </button>
      </div>
    </div>
  );
};

const Flashes = ({ flashes }: FlashesProps) => {
  const [flashesWithIds, setFlashesWithIds] = useState<FlashProps[]>([]);

  const removeFlash = (flashId: string) =>
    setFlashesWithIds((items) => items.filter((item) => item.id !== flashId));

  useEffect(() => {
    setFlashesWithIds((state) => {
      return [
        ...state,
        ...(flashes?.items ?? []).map((flash) => ({
          ...flash,
          id: Math.random().toString(36).slice(2),
          removed: false,
        })),
      ];
    });
  }, [flashes]);

  return (
    <div className="ui-flash" data-id={FLASH_DATA_ID}>
      {flashesWithIds
        .filter((item) => !item.removed)
        .map((flash) => (
          <Flash key={flash.id} {...flash} removeFlash={removeFlash} />
        ))}
    </div>
  );
};

const BackendFlashes = ({ flashes }: BackendFlashesProps) => {
  useEffect(() => {
    const transformedFlashes =
      flashes.map((flash) => {
        const [type, content] = flash;
        return { type, content };
      }) || [];

    if (transformedFlashes.length > 0) {
      const store = getRemoteDataStore();

      store.dispatch({
        type: "flash/push",
        payload: transformedFlashes,
      });
    }
  }, []);

  const WrappedFlashes = ConnectStateWrapper(Flashes, {
    flashes: selectFlashes,
  });

  return <WrappedFlashes />;
};

export { reducerFlashes, FLASH_DATA_ID, Flashes };
export default BackendFlashes;
