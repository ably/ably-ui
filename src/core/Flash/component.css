@layer components {
  .ui-flash {
    @apply w-full fixed;
    top: 5.5rem;
    z-index: calc(var(--stacking-context-page-meganav) - 10);
    transition: margin-top 200ms;
  }

  .ui-flash-message {
    @apply font-sans font-light antialiased max-w-screen-xl mx-auto mt-8 opacity-0 relative;
    transition:
      opacity 200ms,
      transform 200ms,
      height 200ms 200ms,
      margin-top 200ms 200ms;
    transform: translateY(-200%) rotateX(-90deg);
  }

  /* dynamic content inside flash, can't add classes */
  .ui-flash-text a {
    @apply underline;
  }

  .ui-flash-message-enter {
    @apply opacity-100;
    transform: translateY(0) rotateX(0);
  }
}
