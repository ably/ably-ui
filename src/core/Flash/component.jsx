import React, { useEffect, useState, useRef } from "react";
import DOMPurify from "dompurify";
import T from "prop-types";
import { nanoid } from "nanoid/non-secure";

import { getRemoteDataStore } from "../remote-data-store";
import ConnectStateWrapper from "../ConnectStateWrapper/component.jsx";
import Icon from "../Icon/component.jsx";

const REDUCER_KEY = "flashes";
const FLASH_DATA_ID = "ui-flashes";

const initialState = { items: [] };

const reducerFlashes = {
  [REDUCER_KEY]: (state = initialState, action) => {
    switch (action.type) {
      case "flash/push": {
        const flashes = Array.isArray(action.payload) ? action.payload : [action.payload];
        return { items: [...state.items, ...flashes] };
      }
      default:
        return state;
    }
  },
};

const selectFlashes = (store) => store.getState()[REDUCER_KEY];

const FlashT = {
  type: T.oneOf(["error", "success", "notice", "info", "alert"]),
  content: T.string,
};

const FLASH_BG_COLOR = {
  error: "bg-gui-error",
  success: "bg-zingy-green",
  notice: "bg-electric-cyan",
  info: "bg-electric-cyan",
  alert: "bg-active-orange",
};

const FLASH_TEXT_COLOR = {
  error: "text-white",
  success: "text-cool-black",
  notice: "text-cool-black",
  info: "text-cool-black",
  alert: "text-white",
};

const AUTO_HIDE = ["success", "info", "notice"];
const AUTO_HIDE_TIME = 8000;

const useAutoHide = (type, closeFlash) => {
  const timeoutId = useRef(null);

  useEffect(() => {
    if (AUTO_HIDE.includes(type)) {
      timeoutId.current = setTimeout(() => {
        closeFlash();
      }, AUTO_HIDE_TIME);
    }

    return () => {
      if (timeoutId.current) {
        clearTimeout(timeoutId.current);
      }
    };
  }, []);
};

const Flash = ({ id, type, content, removeFlash }) => {
  const ref = useRef(null);
  const [closed, setClosed] = useState(false);
  const [flashHeight, setFlashHeight] = useState(0);
  const [triggerEntryAnimation, setTriggerEntryAnimation] = useState(false);

  const closeFlash = () => {
    if (ref.current) {
      setFlashHeight(ref.current.getBoundingClientRect().height);
    }

    setClosed(true);

    setTimeout(() => {
      removeFlash(id);
    }, 100);
  };

  useEffect(() => setTriggerEntryAnimation(true), []);
  useAutoHide(type, closeFlash);

  const animateEntry = triggerEntryAnimation && !closed;

  let style;

  if (flashHeight && !closed) {
    style = { height: `${flashHeight}px` };
  } else if (closed) {
    style = { height: 0, marginTop: 0, zIndex: -1 };
  } else {
    style = {};
  }

  const safeContent = DOMPurify.sanitize(content, {
    ALLOWED_TAGS: ["a"],
    ALLOWED_ATTR: ["href", "data-method", "rel"],
  });

  const withIcons = {
    notice: "icon-gui-ably-badge",
    success: "icon-gui-tick",
    error: "icon-gui-warning",
    alert: "icon-gui-warning",
  };

  const iconColor = {
    notice: "text-cool-black",
    success: "text-cool-black",
    error: "text-white",
    alert: "text-white",
  };

  return (
    <div className={`ui-flash-message ui-grid-px ${animateEntry ? "ui-flash-message-enter" : ""}`} style={style} ref={ref} data-id="ui-flash">
      <div className={`${FLASH_BG_COLOR[type]} p-32 flex align-center rounded shadow-container-subtle`}>
        {withIcons[type] && <Icon name={withIcons[type]} color={iconColor[type]} size="1.5rem" additionalCSS="mr-16 self-baseline" />}
        <p className={`ui-flash-text ${FLASH_TEXT_COLOR[type]}`} dangerouslySetInnerHTML={{ __html: safeContent }} />
        <button type="button" className="p-0 ml-auto self-start focus:outline-none" onClick={closeFlash}>
          <Icon name="icon-gui-close" color={iconColor[type]} size="1.5rem" additionalCSS="transition-colors" />
        </button>
      </div>
    </div>
  );
};

Flash.propTypes = {
  ...FlashT,
};

const Flashes = ({ flashes }) => {
  const [flashesWithIds, setFlashesWithIds] = useState([]);

  const removeFlash = (flashId) => setFlashesWithIds((items) => items.filter((item) => item.id !== flashId));

  useEffect(() => {
    setFlashesWithIds((state) => {
      return [...state, ...(flashes?.items || []).map((flash) => ({ ...flash, id: nanoid(), removed: false }))];
    });
  }, [flashes]);

  return (
    <div className="ui-flash" data-id={FLASH_DATA_ID}>
      {flashesWithIds
        .filter((item) => !item.removed)
        .map((flash) => (
          <Flash removeFlash={removeFlash} key={flash.id} {...flash} />
        ))}
    </div>
  );
};

Flashes.propTypes = {
  flashes: T.shape({ items: T.arrayOf(T.shape(FlashT)) }),
};

const BackendFlashes = ({ flashes }) => {
  useEffect(() => {
    const transformedFlashes =
      flashes.map((flash) => {
        const [type, content] = flash;
        return { type, content };
      }) || [];

    if (transformedFlashes.length > 0) {
      const store = getRemoteDataStore();

      store.dispatch({
        type: "flash/push",
        payload: transformedFlashes,
      });
    }
  }, []);

  const WrappedFlashes = ConnectStateWrapper(Flashes, { flashes: selectFlashes });

  return <WrappedFlashes />;
};

BackendFlashes.propTypes = {
  flashes: T.arrayOf(T.arrayOf(T.string)),
};

export { reducerFlashes, FLASH_DATA_ID, Flashes };
export default BackendFlashes;
