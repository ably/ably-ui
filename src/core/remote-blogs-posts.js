import { isJsonResponse } from "./remote-data-util";

const fetchBlogPosts = async (store, blogUrl) => {
  try {
    if (!blogUrl) {
      console.log(
        `Skipping fetching blog posts, invalid blogUrl: "${blogUrl}"`
      );
      return;
    }

    const res = await fetch(blogUrl);

    if (isJsonResponse(res.headers.get("content-type"))) {
      const payload = await res.json();
      store.dispatch({ type: "blog/loaded", payload });
    } else {
      throw new Error("Blog posts url is not serving json");
    }
  } catch (e) {
    console.warn("Could not fetch blog posts due to error:", e);
  }
};

const initialState = { recent: null };

const REDUCER_KEY = "blogPosts";

const reducerBlogPosts = {
  [REDUCER_KEY]: (state = initialState, action) => {
    switch (action.type) {
      case "blog/loaded":
        return { ...state, recent: action.payload };
      default:
        return state;
    }
  },
};

const selectRecentBlogPosts = (store) => store.getState()[REDUCER_KEY]?.recent;

export { fetchBlogPosts, reducerBlogPosts, selectRecentBlogPosts };
