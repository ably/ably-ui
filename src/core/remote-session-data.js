// Fetches current users session data
// Assumes an authenticated session, so will only work when used on ably.com/ably.io

import { isJsonResponse } from "./remote-data-util";

const NOT_FOUND_ERROR_CODE = "not-found";

const fetchSessionData = async (store, sessionUrl) => {
  const sessionLoaded = (payload = {}) =>
    store.dispatch({ type: "session/loaded", payload });

  try {
    if (!sessionUrl) {
      console.log(
        `Skipping fetching session, invalid sessionUrl: "${sessionUrl}"`,
      );
      sessionLoaded();
      return;
    }

    const res = await fetch(sessionUrl, { cache: "no-cache" });
    const jsonResponse = isJsonResponse(res.headers.get("content-type"));

    if (!jsonResponse) {
      throw new Error("Session endpoint is not serving json");
    }

    const payload = await res.json();

    if (payload.error === NOT_FOUND_ERROR_CODE) {
      sessionLoaded();
    } else {
      sessionLoaded(payload);
    }
  } catch (e) {
    sessionLoaded();
    console.warn("Could not fetch session data due to error:", e);
  }
};

const initialState = { data: null };

const REDUCER_KEY = "session";

const reducerSessionData = {
  [REDUCER_KEY]: (state = initialState, action) => {
    switch (action.type) {
      case "session/loaded":
        return { ...state, data: action.payload };
      default:
        return state;
    }
  },
};

const selectSessionData = (store) => store.getState()[REDUCER_KEY]?.data;

export { fetchSessionData, reducerSessionData, selectSessionData };
