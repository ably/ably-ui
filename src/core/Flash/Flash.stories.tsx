import React from "react";
import Flash, { reducerFlashes } from "../Flash";
import { expect, within } from "@storybook/test";

import {
  attachStoreToWindow,
  createRemoteDataStore,
} from "../remote-data-store.js";
import { reducerBlogPosts } from "../remote-blogs-posts.js";
import { reducerSessionData } from "../remote-session-data.js";

export default {
  title: "Components/Flash",
  component: Flash,
  args: {
    flashes: [
      ["success", "Congratulations! You've won the Oscar"],
      ["notice", "This is a notice"],
      ["error", "This is an error, very bad"],
      ["alert", "This is an alert"],
      ["info", "Some useful information, you are welcome"],
    ],
  },
};

export const Default = {
  render: (args) => {
    const store = createRemoteDataStore({
      ...reducerBlogPosts,
      ...reducerSessionData,
      ...reducerFlashes,
    });

    attachStoreToWindow(store);

    return <Flash {...args} />;
  },
};

export const WithLinks = {
  args: {
    flashes: [
      ["success", 'Valid link: <a href="/valid-link">Click me</a>'],
      [
        "error",
        'Invalid link: <a href="https://external.com">Should be removed</a>',
      ],
      [
        "info",
        'Link with data-method: <a href="/delete" data-method="delete">Delete</a>',
      ],
      [
        "notice",
        'Invalid link with proto relative: <a href="//external.com" rel="noopener">External</a>',
      ],
    ],
  },
  render: (args) => {
    const store = createRemoteDataStore({
      ...reducerBlogPosts,
      ...reducerSessionData,
      ...reducerFlashes,
    });

    attachStoreToWindow(store);

    return <Flash {...args} />;
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);

    // Wait for flashes to appear
    const flashes = await canvas.findAllByTestId("ui-flash");

    // Test valid link (should be present)
    const validLinkFlash = flashes[0];
    const validLink = within(validLinkFlash).getByText("Click me");
    expect(validLink).toBeInTheDocument();
    expect(validLink).toHaveAttribute("href", "/valid-link");

    // Test invalid link (should be removed)
    const invalidLinkFlash = flashes[1];
    const invalidLinkText =
      within(invalidLinkFlash).getByText("Should be removed");
    expect(invalidLinkText).toBeInTheDocument();
    expect(invalidLinkText).not.toHaveAttribute("href");

    // Test link with data-method (should be present)
    const dataMethodFlash = flashes[2];
    const dataMethodLink = within(dataMethodFlash).getByRole("link", {
      name: "Delete",
    });
    expect(dataMethodLink).toBeInTheDocument();
    expect(dataMethodLink).toHaveAttribute("href", "/delete");
    expect(dataMethodLink).toHaveAttribute("data-method", "delete");

    // Test link with proto relative url (should be present)
    const relFlash = flashes[3];
    const relLink = within(relFlash).getByText("External");
    expect(relLink).toBeInTheDocument();
    expect(relLink).not.toHaveAttribute("href");
    expect(relLink).not.toHaveAttribute("rel");
  },
};
