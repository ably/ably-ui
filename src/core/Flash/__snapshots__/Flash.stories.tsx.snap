// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Flash Default smoke-test 1`] = `
<div class="ui-flash"
     data-id="ui-flashes"
>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-zingy-green p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-cool-black mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-check-outline">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-cool-black">
        Congratulations! You've won the Oscar
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-cool-black transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-electric-cyan p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-cool-black mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-ably-badge">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-cool-black">
        This is a notice
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-cool-black transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-gui-error p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-white mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-exclamation-triangle-outline">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-white">
        This is an error, very bad
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-white transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-active-orange p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-white mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-exclamation-triangle-outline">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-white">
        This is an alert
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-white transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-electric-cyan p-32 flex align-center rounded shadow-container-subtle">
      <p class="ui-text-p1 mr-16 text-cool-black">
        Some useful information, you are welcome
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
      </button>
    </div>
  </div>
</div>
`;

exports[`Components/Flash WithLinks play-test 1`] = `
<div class="ui-flash"
     data-id="ui-flashes"
>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-zingy-green p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-cool-black mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-check-outline">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-cool-black">
        Valid link:
        <a href="/valid-link">
          Click me
        </a>
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-cool-black transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-gui-error p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-white mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-exclamation-triangle-outline">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-white">
        Invalid link:
        <a>
          Should be removed
        </a>
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-white transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-electric-cyan p-32 flex align-center rounded shadow-container-subtle">
      <p class="ui-text-p1 mr-16 text-cool-black">
        Link with data-method:
        <a data-method="delete"
           href="/delete"
        >
          Delete
        </a>
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
      </button>
    </div>
  </div>
  <div class="ui-flash-message ui-grid-px ui-flash-message-enter"
       data-id="ui-flash"
       data-testid="ui-flash"
  >
    <div class="bg-electric-cyan p-32 flex align-center rounded shadow-container-subtle">
      <svg class="text-cool-black mr-16 self-baseline"
           style="width: 1.5rem; height: 1.5rem;"
      >
        <use xlink:href="#sprite-icon-gui-ably-badge">
        </use>
      </svg>
      <p class="ui-text-p1 mr-16 text-cool-black">
        Invalid link with proto relative:
        <a>
          External
        </a>
      </p>
      <button type="button"
              class="p-0 ml-auto self-start focus:outline-none"
      >
        <svg class="text-cool-black transition-colors"
             style="width: 1.5rem; height: 1.5rem;"
        >
          <use xlink:href="#sprite-icon-gui-x-mark-outline">
          </use>
        </svg>
      </button>
    </div>
  </div>
</div>
`;
